api4/channel.go

```diff
@@ -1802,6 +1805,86 @@ func removeChannelMember(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func addChannelPreviewer(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	props := model.StringInterfaceFromJSON(r.Body)
+	userId, ok := props["user_id"].(string)
```

Comment:
model/utils has a `MapFromJSON` method which does the same thing without needing a type assertion every time.